### ISSUE TITLE
MSFragger Exit Code 152 Linux Headless on Computing Cluster

### ISSUE THREAD

**Issue by chad-hyer** [Nesvilab/FragPipe#1733]
Hey. I'm setting up FragPipe to run headless on my institution's Linux cluster and am running into some issues with unexplained errors. I have run this same exact workflow on a different Linux computer headless with no errors and also in the GUI form on Windows. Originally, I was having problems with permissions and gave all the files execute permissions so that FragPipe and the tools would even run on the cluster. This has brought me to the most recent error, and I can't find any documentation about what is going wrong. I get an exit code of 152 from MSFragger with no explanation of what that error is. Attached is the log. Please advise.

[log.txt](https://github.com/user-attachments/files/16642597/log.txt)

**Comment by fcyu** [comment]
Were you running FragPipe in a Docker container? The exit code 152 seems related to some issues in the container.

Best,

Fengchao

**Comment by chad-hyer** [comment]
I don't believe it should be running in a docker container. I believe the tools folder I pulled onto the cluster comes from the jre version I had installed on my windows machine. I didn't try to set it up as a docker container at any point.

**Comment by fcyu** [comment]
Thanks for the clarify. May I ask how did you get/convert the mzML file? Because MSFragger couldn't detected the analyzer: `/grphome/grp_FragPipe/fragpipe/Test/inputs/5MQ-0.mzML: Scans = 6278; ITMS: false; FTMS: false; Isolation sizes = []`. Not sure if it is related. If it is OK, could you share one of the mzML files with us?

Thanks,

Fengchao

**Comment by chad-hyer** [comment]
The mzMLs were generated using MS Convert from .d files from an Agilent 6560. Here is a link to one of the mzML files: https://drive.google.com/file/d/1VliSsU7JJRDyhQDiRhO5ylidEr9LEoSb/view?usp=sharing

**Comment by fcyu** [comment]
Thank you for sharing your file. FragPipe can process the file without any issues on my Windows desktop. Without further information, I am afraid I can't figure out the reason of the error.

Best,

Fengchao

**Comment by chad-hyer** [comment]
Thanks for the help. Is there any information that I could provide that would make it possible to find the issue? I will continue to play with it on my linux pc and see if uploading my build from there fixes the issue. I wouldn't be surprised if some of the issues come from permissions issues on our cluster, so I will reach out to them for support as well.

Thanks,
Chad

**Comment by fcyu** [comment]
The error code was from either JVM or system level, thus I am not sure what kind of information can help me to troubleshoot it.

BTW, is the error reproducible?

Best,

Fengchao

**Comment by chad-hyer** [comment]
Gotcha. I have only tested it using these mzML's and have gotten the same error each time I have loaded it. I will try it on some other data in the coming days to see if it has the same issue, but I suspect the issue is coming from the way the cluster manages permissions.

**Comment by chad-hyer** [comment]
Quick question. You mentioned the error is related to docker. I just tested out the non-jre linux version and got the same error. Are any of the tools built using docker by default? My cluster has had issues with docker based programs in the past.

**Comment by fcyu** [comment]
>  I just tested out the non-jre linux version and got the same error. 

This `jre` is not related to docker. It indicates if the zip file contains the Java runtime **for Windows**.

> Are any of the tools built using docker by default?

No, just we have a docker image besides the zip files for people to pull.

Have you tried the zip file without `jre`?

Best,

Fengchao

**Comment by chad-hyer** [comment]
Yeah, I tried bother the jre and non-jre versions. I was able to run both locally on my linux computer using all the same settings and files as on the cluster but only get the error on the cluster. I'm going to talk to my cluster administrator about why he thinks it that is. Do you know which files would yield exit code 152? That should give me more information to provide him when we try to troubleshoot it.

**Comment by fcyu** [comment]
It is `fragpipe/tools/MSFragger-4.1/MSFragger-4.1.jar`. You could try to reproduce it using the following command if you want.

```shell
java -jar -Dfile.encoding=UTF-8 -Xmx616G /grphome/grp_FragPipe/fragpipe/tools/MSFragger-4.1/MSFragger-4.1.jar /grphome/grp_FragPipe/fragpipe/Test/outputs/Test1/fragger.params /grphome/grp_FragPipe/fragpipe/Test/inputs/5MQ-0.mzML /grphome/grp_FragPipe/fragpipe/Test/inputs/5MQ-1.mzML /grphome/grp_FragPipe/fragpipe/Test/inputs/5MQ-2.mzML /grphome/grp_FragPipe/fragpipe/Test/inputs/5MQ-3.mzML /grphome/grp_FragPipe/fragpipe/Test/inputs/5MQ-4.mzML /grphome/grp_FragPipe/fragpipe/Test/inputs/5MQ-5.mzML /grphome/grp_FragPipe/fragpipe/Test/inputs/5MQ-6.mzML /grphome/grp_FragPipe/fragpipe/Test/inputs/5MQ-7.mzML /grphome/grp_FragPipe/fragpipe/Test/inputs/5MQ-8.mzML /grphome/grp_FragPipe/fragpipe/Test/inputs/5MQ-9.mzML
```

Best,

Fengchao

**Comment by chad-hyer** [comment]
Thanks! I'll try that tomorrow and see what the log gives me

**Comment by chad-hyer** [comment]
I figured out the problem. The issue was with how the cluster was allocating memory to the task. Setting the ram and threads in the command fixed the issue. It ran into a new issue with using Alpha_pept_RT from a koina server that was fixed using DIA-NN instead. I would still like to be able to use other MSBooster koina models if possible. My guess is that the error occurs when the cluster tries to use the curl command to get the RT model. Is there potentially a way to have fragpipe use a local RT model instead of trying to pull from the koina server each time? Attached is the error from the output.
[slurm-66398944.txt](https://github.com/user-attachments/files/16700570/slurm-66398944.txt)

**Comment by fcyu** [comment]
Glad to hear that you figured out the reason of the issue.

> My guess is that the error occurs when the cluster tries to use the curl command to get the RT model.

We are aware of this issue and has replaced curl with the Java native HTTPS. Let me know if you want to test the pre-released version.

> Is there potentially a way to have fragpipe use a local RT model instead of trying to pull from the koina server each time?

Since the prediction models other than DIA-NN and MS2PIP require GPU and some Python package, I don't think it is easy for people to run these models locally. This is why we applied Koina server in MSBooster.

Best,

Fengchao

**Comment by chad-hyer** [comment]
The pre-released version would be great. I'm happy to help test it!

**Comment by fcyu** [comment]
Sounds good. Here is the link to download the pre-released FragPipe: https://www.dropbox.com/t/bTGxKu9E6gh0ySS8

Best,

Fengchao

**Comment by chad-hyer** [comment]
Thanks, I tried it out and got an error. For reference, I am only using the tools folder from the pre-release with a copy of the rest of the directories from the older release (just in case this is contributing to the problem). It looks like my cluster may just have issues with connecting to the koina server, so it may not necessarily be a problem with MSBooster. I've attached the output.
[slurm-66403691.txt](https://github.com/user-attachments/files/16717350/slurm-66403691.txt)
